libopenage/unit/producer.cpp

```diff
@@ -564,6 +569,14 @@ void BuildingProducer::initialise(Unit *unit, Player &player) {
 	unit->add_attribute(std::make_shared<Attribute<attr_type::hitpoints>>(this->unit_data.hit_points));
 	unit->add_attribute(std::make_shared<Attribute<attr_type::damaged>>(this->unit_data.hit_points));
 
+	// population
+	if (this->id() == 109 || this->id() == 70) { // Town center, House
+		unit->add_attribute(std::make_shared<Attribute<attr_type::population>>(-5));
+	}
+	else if (this->id() == 82) { // Castle
+		unit->add_attribute(std::make_shared<Attribute<attr_type::population>>(-20));
+	}
```

Comment:
maybe you can get the exact amount from the gamedata?
https://github.com/SFTtech/openage/blob/5f6d4f2078b5a14d312041e14a8d8062f0b224c4/openage/convert/gamedata/unit.py#L233
But that's probably not the values that buildings grant, instead it seems that it's the slot price of a unit.